lua/devcontainer-cli/devcontainer_utils.lua

```diff
@@ -27,64 +55,89 @@ local on_fail = function(exit_code)
 ---@param code (integer) the exit code
-local on_exit = function(_, code, _)
+local _on_exit = function(code)
   if code == 0 then
-    on_success()
+    _on_success()
     return
   end
 
-  on_fail(code)
+  _on_fail(code)
 end
 
---- execute command
----@param cmd (string) the command to execute in the devcontainer terminal
-function M.exec_command(cmd)
-  vim.fn.termopen(
-    cmd,
-    {
-      on_exit = on_exit,
-      on_stdout = function(_, _, _)
-        if win ~= -1 then
-          vim.api.nvim_win_call(
-            win,
-            function()
-              vim.cmd("normal! G")
-            end
-          )
-        else
-          vim.notify("Executed " .. cmd)
-        end
-      end,
-    }
-  )
-  if buffer ~= -1 then
-    vim.api.nvim_set_current_buf(buffer)
-  else
-    vim.notify("No buffer created, therefore no output from command will be visible", vim.log.levels.WARN)
+-- check if the value is in the given table
+local function tableContains(tbl, value)
+  for _, item in ipairs(tbl) do
+    if item == value then
+      return true
+    end
   end
+
+  return false
 end
 
 -- create a new window and execute the given command
 ---@param cmd (string) the command to execute in the devcontainer terminal
-local function spawn_and_execute(cmd)
-  prev_win = vim.api.nvim_get_current_win()
-  win, buffer = windows_utils.open_floating_window(on_detach)
-  M.exec_command(cmd)
+---@param direction (string|nil) the placement of the window to be created (float, horizontal, vertical)
+---@param close_on_exit (boolean|nil) whether to close terminal when process exits
+local function _spawn_and_execute(cmd, direction, close_on_exit)
+  direction = vim.F.if_nil(direction, "float")
+  if tableContains(directions, direction) == false then
+    vim.notify("Invalid direction: " .. direction, vim.log.levels.ERROR)
+    return
+  end
+
+  -- create the terminal
+  _terminal = Terminal:new {
+    cmd = cmd,
+    hidden = false,
+    display_name = "devcontainer-cli",
+    direction = vim.F.if_nil(direction, "float"),
+    dir = folder_utils.get_root(config.toplevel),
+    close_on_exit = vim.F.if_nil(close_on_exit, false),
+    on_open = function(term)
+      -- ensure that we are not in insert mode
+      vim.cmd("stopinsert")
+      vim.api.nvim_buf_set_keymap(
+        term.bufnr,
+        'n',
+        '<esc>',
+        '<CMD>lua vim.api.nvim_buf_delete(' .. term.bufnr .. ', { force = true } )<CR><CMD>close<CR>',
+        { noremap = true, silent = true }
+      )
+      vim.api.nvim_buf_set_keymap(
+        term.bufnr,
+        'n',
+        'q',
+        '<CMD>lua vim.api.nvim_buf_delete(' .. term.bufnr .. ', { force = true } )<CR><CMD>close<CR>',
+        { noremap = true, silent = true }
+      )
+      vim.api.nvim_buf_set_keymap(term.bufnr, 'n', 't', '<CMD>close<CR>', { noremap = true, silent = true })
+    end,
+    auto_scroll = true,
+    on_exit = function(_, _, code, _)
+      _on_exit(code)
+      _on_detach()
```

Comment:
enhancement: Currently when a window is hidden and the process exits the terminal window also exits. It would be nice if the window persisted until it was closed by the user.

Reply:
Created issue #14